File: queries/oferta.js
const client = require('../client');

function all() {
  const query = `select * from oferta;`;
  return client.query(query)
}

function select(numserie) {
  const query = `select oferta.* from oferta
                where numSerie = ${numserie};`;
  return client.query(query)
}

function selectBy(key, value) {
  const query = `select oferta.* from oferta
                  where ${key} = ${value};`;
  return client.query(query)
}

function insert(oferta) {
  const columns = Object.keys(oferta);
  let values = Object.values(oferta);

  values = values.map(function (oferta) {
    return `'${value}'`;
  })

  const query = `insert into oferta (${columns.join(',')})
                  values (${values.join(',')})
                  returning *;`;

  return client.query(query);
}

function update(numserie, oferta) {
  const entries = Object.entries(oferta);

  let query = `update oferta set `;

  entries.forEach(function (key, index) {
    query += `${key[0]} = '${key[1]}'`;

    if (index < entries.length - 1) {
      query += `,`;
    }
  });

  query += `where numSerie = ${numserie} returning *;`;
  return client.query(query);
}

function del(numserie) {
  const query = `delete from oferta where numSerie = ${numserie} returning *`;

  return client.query(query);
}

  async function buscarOferta(data){
  
    const query = `select * from oferta where puesto = '${data.puesto}' AND area = '${data.area}'`;
    console.log(query);
     /*router.post('/buscaroferta',function(req,res){
      res.sendFile (__dirname+'/Inicio.html');
      console.log(req.body);
    });*/

    client.query(query, (error, result) => {
      console.log(result)
    })

    let result = await client.query(query);

    console.log('result', result.rows);
    

    //res.render('listarofertas', { oferta: filas });

    console.log(query)
  

    return result.rows;
  }

  
  function insertOferta(data){
    
      const query = `insert into oferta(numSerie, sueldo, requisitos, area, puestosVacantes, descripcion, edadRequerida, puesto)
                      values('${data.numSerie}','${data.sueldo}','${data.requisitos}','${data.area}','${data.puestosVacantes}',
                      '${data.descripcion}', '${data.edadRequerida}','${data.puesto}')`;
    
      console.log(query)
    
      return client.query(query);
    }

module.exports = {
  all,
  select,
  selectBy,
  insert,
  update,
  del,
  buscarOferta,
  insertOferta,
};

File: Anijobs0/Clases/Oferta.vb
Public Class Oferta
    Dim _numSerie As Integer
    Dim _puesto As String
    Dim _descripcion As String
    Dim _disponibilidad As String
    Dim _requisitos As String
    Dim _sueldo As Integer
    Dim _area As String
    Dim _edadRequerida As String

    Public Property NumSerie As Integer
        Get
            Return _numSerie
        End Get
        Set(value As Integer)
            _numSerie = value
        End Set
    End Property

    Public Property Puesto As String
        Get
            Return _puesto
        End Get
        Set(value As String)
            _puesto = value
        End Set
    End Property

    Public Property Descripcion As String
        Get
            Return _descripcion
        End Get
        Set(value As String)
            _descripcion = value
        End Set
    End Property

    Public Property Disponibilidad As String
        Get
            Return _disponibilidad
        End Get
        Set(value As String)
            _disponibilidad = value
        End Set
    End Property

    Public Property Requisitos As String
        Get
            Return _requisitos
        End Get
        Set(value As String)
            _requisitos = value
        End Set
    End Property

    Public Property Sueldo As Integer
        Get
            Return _sueldo
        End Get
        Set(value As Integer)
            _sueldo = value
        End Set
    End Property

    Public Property Area As String
        Get
            Return _area
        End Get
        Set(value As String)
            _area = value
        End Set
    End Property

    Public Property EdadRequerida As String
        Get
            Return _edadRequerida
        End Get
        Set(value As String)
            _edadRequerida = value
        End Set
    End Property

    'Constructor'
    Sub New(nserie As Integer, psto As String, desc As String, disp As String, req As String, sdo As Integer, ara As String, edad As String)
        NumSerie = nserie
        Puesto = psto
        Descripcion = desc
        Disponibilidad = disp
        Requisitos = req
        Sueldo = sdo
        Area = ara
        EdadRequerida = edad
    End Sub
End Class
